### PR TITLE
[Merged by Bors] - refactor(LinearAlgebra): Ensure `ChooseBasisIndex` is finite on trivial modules

### PR DESCRIPTION
**Pull request by eric-wieser**
This also changes `basisFintypeOfFiniteSpans` to use `Finite` rather than `Fintype`, as it was noncomputable anyway.
This means it has to be renamed to `basis_finite_of_finite_spans` as it now is a proof!

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/LinearAlgebra/Dual.lean [hunk 1]
```diff
@@ -566,16 +566,15 @@ theorem forall_dual_apply_eq_zero_iff (v : V) : (∀ φ : Module.Dual K V, φ v
...
-  (Basis.ofVectorSpace K V).dual_rank_eq
+  (Module.Free.chooseBasis K V).dual_rank_eq
 #align module.dual_rank_eq Module.dual_rank_eq
 
 -- Porting note: broken dot notation lean4#1910 LinearMap.range
-theorem erange_coe [FiniteDimensional K V] : LinearMap.range (eval K V) = ⊤ :=
-  letI : IsNoetherian K V := IsNoetherian.iff_fg.2 inferInstance
-  (Basis.ofVectorSpace K V).eval_range
+theorem erange_coe [Module.Finite K V] : LinearMap.range (eval K V) = ⊤ :=
+  -- letI : IsNoetherian K V := IsNoetherian.iff_fg.2 inferInstance
```
Comment (ocfnash) [inline Mathlib/LinearAlgebra/Dual.lean:576]:
```suggestion
```

**Review by ocfnash — COMMENTED** [review]
Thanks very much for doing this. Lots of small wins here.

bors d+

**Comment by eric-wieser** [comment]
bors merge

**Comment by eric-wieser** [comment]
bors merge

**Pull request closed without merging**